contrib/doca_urom_ucc_plugin/DOCA_UROM_README.md

```diff
@@ -0,0 +1,145 @@
+# Plugin Description
+The CL_DOCA_UROM plugin enables collective offloads via DOCA and CL_DOCA_UROM feature. The plugin runs on the DPU and is initialized via the command channel from CL_DOCA_UROM. It leverages UCC collectives optimized for the DPU and facilitates efficient computation and communication overlap. While the plugin supports all UCC collectives via a copy-in/out method, the optimized algorithms are Allreduce and Alltoall/v (OpenSHMEM).
+
+## Overall Design and System Architecture
```

Comment:
@janjust  Let's change this heading to components and dependencies?

Reply:
done